tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py

```diff
@@ -0,0 +1,207 @@
+        extra_ag_args = {
+            'valid_stacker': False,
+            'problem_types': [BINARY, MULTICLASS],  # Does not support regression
+        }
+        default_ag_args.update(extra_ag_args)
+        return default_ag_args
+
+    def _preprocess(self, X, fit=False, **kwargs):
+        X = super()._preprocess(X, **kwargs)
+        if fit:
+            X_features = list(X.columns)
+            if len(X_features) != 1:
+                raise AssertionError(f'ImagePredictorModel only supports one image feature, but {len(X_features)} were given: {X_features}')
+            if X_features[0] != 'image':
+                self._internal_feature_map = {X_features[0]: 'image'}
+        if self._internal_feature_map:
+            X = X.rename(columns=self._internal_feature_map)
+        return X
+
+    def _fit(self,
+             X: pd.DataFrame,
+             y: pd.Series,
+             X_val: Optional[pd.DataFrame] = None,
+             y_val: Optional[pd.Series] = None,
+             time_limit: Optional[int] = None,
+             sample_weight=None,
+             verbosity=2,
+             **kwargs):
+        """The internal fit function
+
+        Parameters
+        ----------
+        X
+            Features of the training dataset
+        y
+            Labels of the training dataset
+        X_val
+            Features of the validation dataset
+        y_val
+            Labels of the validation dataset
+        time_limit
+            The time limit for the fit function
+        kwargs
+            Other keyword arguments
+
+        """
+        try_import_mxnet()
+        try_import_autogluon_vision()
+        from autogluon.vision import ImagePredictor, ImageDataset
+        params = self._get_model_params()
+
+        X = self.preprocess(X, fit=True)
+        if X_val is not None:
+            X_val = self.preprocess(X_val)
+
+        if sample_weight is not None:  # TODO: support
+            logger.log(15, "sample_weight not yet supported for ImagePredictorModel, this model will ignore them in training.")
+
+        X = X.reset_index(drop=True)
+        y = y.reset_index(drop=True)
+        if X_val is not None:
+            X_val = X_val.reset_index(drop=True)
+            y_val = y_val.reset_index(drop=True)
+        X[self._label_column_name] = y
+        if X_val is not None:
+            X_val[self._label_column_name] = y_val
+
+        verbosity_image = max(0, verbosity - 1)
+        root_logger = logging.getLogger()
+        root_log_level = root_logger.level
+        # TODO: ImagePredictor doesn't use problem_type in any way at present.
+        #  It also doesn't error or warn if problem_type is not one it expects.
+        self.model = ImagePredictor(
+            problem_type=self.problem_type,
+            path=self.path,
+            # eval_metric=self.eval_metric,  # TODO: Vision only works with accuracy
+            verbosity=verbosity_image
+        )
+        classes = list(y.unique())
```

Comment:
We may need to call the following to avoid the case that some rare classes only appear in the validation set:

```python
if X_val is None:
    classes = list(y.unique())
else:
    classes = list(pd.concat([y, y_val]).unique())
```

Reply:
Good point, added